### PR TITLE
Change type to prevent overflow and negative value.

### PR DESCRIPTION
**Pull request by SergiiChurakov**
## Description of the changes
Change type to prevent overflow and negative value.

## Reference/Link to the issue solved with this PR (if any)
#969

### CODE REVIEW

File: iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs [hunk 1]
```diff
@@ -98,8 +98,8 @@ public override ShouldRetry GetShouldRetry()
                 if (currentRetryCount < this.retryCount)
                 {
                     Random random = new Random();
-                    int num = (int)((Math.Pow(2.0, (double)currentRetryCount) - 1.0) * (double)random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2)));
```
Comment (CIPop) [inline iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs]: I don't think this fixes it. Instead of changing the type, I think you need to add `checked { ... }` around the operation.
Please also add a unit test to ensure this will throw on overflow.

**Comment by CIPop** [comment]
@SergiiChurakov no activity for 11 days. Closing for now.
Feel free to resubmit taking into account the observations above.
